core/kazoo_endpoint/src/kz_attributes.erl

```diff
@@ -265,33 +265,33 @@ maybe_get_account_default_number(Number, Name, Account, Call) ->
             maybe_get_assigned_number(Number, Name, Call)
     end.
 
--spec maybe_get_assigned_number(ne_binary(), ne_binary(), kapps_call:call()) ->
+-spec maybe_get_assigned_number(ne_binary(), ne_binary(), ne_binary()|kapps_call:call()) ->
                                        {api_binary(), api_binary()}.
-maybe_get_assigned_number(_, Name, Call) ->
-    AccountDb = kapps_call:account_db(Call),
-    case kz_datamgr:open_cache_doc(AccountDb, ?KNM_PHONE_NUMBERS_DOC) of
-        {'error', _R} ->
-            Number = default_cid_number(),
-            lager:warning("could not open ~s doc <~s> ~s: ~p", [?KNM_PHONE_NUMBERS_DOC, Name, Number, _R]),
-            {Number, Name};
-        {'ok', JObj} ->
-            PublicJObj = kz_json:public_fields(JObj),
+maybe_get_assigned_number(_, Name, ?MATCH_ACCOUNT_ENCODED(_)=AccountDb) ->
+    case knm_numbers:account_listing(AccountDb) of
+        [_|_] = NumbersList ->
             Numbers = [Num
-                       || Num <- kz_json:get_keys(PublicJObj)
+                       || {Num,JObj} <- NumbersList
                               ,Num =/= <<"id">>
-                              ,(not kz_json:is_true([Num, <<"on_subaccount">>], JObj))
-                              ,(kz_json:get_value([Num, <<"state">>], JObj) =:= ?NUMBER_STATE_IN_SERVICE)
+                              ,(not kz_json:is_true(<<"on_subaccount">>, JObj))
```

Comment:
This does not appear in the view. Maybe add it to ./applications/crossbar/priv/couchdb/account/phone_numbers.json ?